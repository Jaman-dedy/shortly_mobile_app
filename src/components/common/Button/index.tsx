import React from 'react';
import {ActivityIndicator} from 'react-native';
import {useTheme} from '@shopify/restyle';

import {BaseButton, Box, Text, Theme} from '@app/assets/style/theme';

export type ButtonProps = {
  onPress: () => void;
  text: string;
  color: 'mainBackground' | 'mainForeground' | 'primary' | 'primaryDark';
  loading?: boolean;
};

const Button = ({text, onPress, color, loading}: ButtonProps) => {
  const theme = useTheme<Theme>();
  return (
    <BaseButton
      onPress={onPress}
      backgroundColor={color}
      marginTop="sm"
      style={theme.buttonVariant}>
      <Box flexDirection="row" justifyContent="center">
        {loading && <ActivityIndicator color="#fff" />}
        <Text variant="buttonText">{text}</Text>
      </Box>
    </BaseButton>
  );
};

export default Button;
