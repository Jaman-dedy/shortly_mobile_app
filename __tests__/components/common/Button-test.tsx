import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup} from '@testing-library/react-native';

import Button from '../../../src/components/common/Button';

import MockThemes from '../../../mock/MockTheme';

afterEach(cleanup);
test('renders correctly', async () => {
  const tree = await renderer
    .create(
      <MockThemes>
        <Button />
      </MockThemes>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
