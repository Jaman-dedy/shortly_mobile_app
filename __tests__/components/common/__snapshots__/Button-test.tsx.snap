// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadiu": 4,
      "marginTop": 8,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "width": 280,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Poppins-Medium",
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    />
  </View>
</View>
`;
